### PR TITLE
GFORMS-814 Added flag to show/hide HMRC logo on all pages and pdfs

### PR DESCRIPTION
**Pull request by girishkamat**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormCategory.scala [hunk 1]
```diff
@@ -20,7 +20,6 @@ import play.api.libs.json._
 
 sealed trait FormCategory {
   def getString: String
-  def isHmrc: Boolean = this != OfstedDefault
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormCategory.scala:23]: So is the plan to replace `"formCategory": "ofstedDefault",`
with this new flag in all our forms?

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormCategory.scala:23]: Yes, the ticket mentions that.

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormCategory.scala:23]: I think the formCategory will remain. however it wont be used to hide/display  hmrc logo

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormTemplate.scala [hunk 2]
```diff
@@ -37,9 +37,10 @@ case class FormTemplate(
   parentFormSubmissionRefs: List[FormComponentId],
   languages: AvailableLanguages,
   save4LaterInfoText: Option[Save4LaterInfoText],
-  summarySection: SummarySection
-) {
+  summarySection: SummarySection,
+  displayHMRCLogo: Option[Boolean]) {
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormTemplate.scala:41]: Can't we this make just a `Boolean`?

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormTemplate.scala:41]: Option[Boolean] reflects the fact that this is indeed an optional value in the template. I was trying to make a simple Boolean, by introducing something similar to  UploadableHmrcDestination, however felt its better to leave it optional. Is there any other way other than using a complementary case class to convert to the main class?

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormTemplate.scala:41]: We don't need to reflect the fact how things are defined in template, if it is not good for our code.

If we want to see template differently (for example as if `displayHMRCLogo` was mandatory) we are preprocessing template in `normaliseJSON` [see here](https://github.com/hmrc/gform/blob/master/app/uk/gov/hmrc/gform/formtemplate/FormTemplatesControllerRequestHandler.scala#L79-L81). With this approach we don't need any complementary case class to achieve that.

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormTemplate.scala]: Thanks Alex, thats a valuable suggestion. I have updated the code accordingly

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**